examples/src/main/java/org/apache/spark/examples/streaming/JavaDirectKerberizedKafkaWordCount.java

```diff
@@ -0,0 +1,107 @@
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.streaming.api.java.*;
+import org.apache.spark.streaming.kafka010.ConsumerStrategies;
+import org.apache.spark.streaming.kafka010.KafkaUtils;
+import org.apache.spark.streaming.kafka010.LocationStrategies;
+import org.apache.spark.streaming.Durations;
+
+/**
+ * Consumes messages from one or more topics in Kafka and does wordcount.
+ * Usage: JavaDirectKerberizedKafkaWordCount <brokers> <groupId> <topics>
+ *   <brokers> is a list of one or more Kafka brokers
+ *   <groupId> is a consumer group name to consume from topics
+ *   <topics> is a list of one or more kafka topics to consume from
+ *
+ * Example:
+ *    $ bin/run-example --files ${path}/kafka_jaas.conf \
+ *      --driver-java-options "-Djava.security.auth.login.config=${path}/kafka_jaas.conf" \
+ *      --conf \
+ *      "spark.executor.extraJavaOptions=-Djava.security.auth.login.config=./kafka_jaas.conf" \
+ *      streaming.JavaDirectKerberizedKafkaWordCount broker1-host:port,broker2-host:port \
+ *      consumer-group topic1,topic2
+ */
+
+public final class JavaDirectKerberizedKafkaWordCount {
+  private static final Pattern SPACE = Pattern.compile(" ");
+
+  public static void main(String[] args) throws Exception {
+    if (args.length < 3) {
+      System.err.println(
+              "Usage: JavaDirectKerberizedKafkaWordCount <brokers> <groupId> <topics>\n" +
+                      "  <brokers> is a list of one or more Kafka brokers\n" +
+                      "  <groupId> is a consumer group name to consume from topics\n" +
+                      "  <topics> is a list of one or more kafka topics to consume from\n\n");
+      System.exit(1);
+    }
+
+    StreamingExamples.setStreamingLogLevels();
+
+    String brokers = args[0];
+    String groupId = args[1];
+    String topics = args[2];
+
+    // Create context with a 2 seconds batch interval
+    SparkConf sparkConf = new SparkConf().setAppName("JavaDirectKerberizedKafkaWordCount");
+    JavaStreamingContext jssc = new JavaStreamingContext(sparkConf, Durations.seconds(2));
+
+    Set<String> topicsSet = new HashSet<>(Arrays.asList(topics.split(",")));
+    Map<String, Object> kafkaParams = new HashMap<>();
+    kafkaParams.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokers);
+    kafkaParams.put(ConsumerConfig.GROUP_ID_CONFIG, groupId);
+    kafkaParams.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+    kafkaParams.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+    kafkaParams.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG,
+                                                  SecurityProtocol.SASL_PLAINTEXT.name);
```

Comment:
`SASL_PLAINTEXT` is only for testing.
Either I would use `SASL_SSL` or log a warning like `SASL_PLAINTEXT is only for testing...`.

Reply:
I'm not questioning whether `SASL_PLAINTEXT` works or not, it is. I'm telling that using kerberos on plain text channel is coming from evil from security perspective. Somehow we should tell the users it's not the advised way because credentials can be sniffed.

Reply:
> I'm not questioning whether `SASL_PLAINTEXT` works or not, it is. I'm telling that using kerberos on plain text channel is coming from evil from security perspective. Somehow we should tell the users it's not the advised way because credentials can be sniffed.

I  agree it's not completely secure. It's just for testing, if only `SASL_PLAINTEXT` used. But, I don't think it's just for testing with kerbreos. 
@dongjoon-hyun @srowen @HyukjinKwon ,can you give some advice.

Reply:
I think you can just explain why `SASL_PLAINTEXT` is discouraged (SSL encryption is disabled and why it's dangerous) in a comment.

Reply:
As I understand it, `SASL_PLAINTEXT` is plaintext, we can get username password and other information easy. But with kerberos, can get location of keytab on server and principal only, you can't get keytab file. So, it's enough. 
SSL encryption and kerberos both cause loss of performance.

Reply:
> SSL encryption and kerberos both cause loss of performance.

As any kind of security solution in the world.

Reply:
@hddong, the purpose of doing Kerberos is for security. If we want more performance over security, there's no point of using Kerberos.

It won't hurt to describe `SASL_PLAINTEXT` vs `SASL_SSL`. You can just say, for example:
This file uses `SASL_PLAINTEXT` for simplicity; however, `SASL_PLAINTEXT` has no SSL encryption and likely be less secure. Consider using `SASL_SSL`.

Feel free to reword and rephrase it ^.

Reply:
> @hddong, the purpose of doing Kerberos is for security. If we want more performance over security, there's no point of using Kerberos.
> 
> It won't hurt to describe `SASL_PLAINTEXT` vs `SASL_SSL`. You can just say, for example:
> This file uses `SASL_PLAINTEXT` for simplicity; however, `SASL_PLAINTEXT` has no SSL encryption and likely be less secure. Consider using `SASL_SSL`.
> 
> Feel free to reword and rephrase it ^.

Thanks, worried about my reading comprehension ability. I will add note for this file.